### ISSUE TITLE
eth: RPC API does not enforce 0x prefix

### ISSUE THREAD

**Issue by gavofyork** [ethereum/go-ethereum#2438]
#### System information

Geth version: All versions
OS & Version: Windows/Linux/OSX
Commit hash : All commits
#### Expected behaviour

Should return: 

```
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params",
    "data": null
  },
  "id": 274
}
```
#### Actual behaviour

Accepts and queues a transaction with the data interpreted as hex.
#### Steps to reproduce the behaviour

Make JSONRPC request using:

```
{"id":253,"jsonrpc":"2.0","method":"eth_sendRawTransaction","params":["f8898085061467a0cc832fd61894b816c3abdbb9967485f8645bc83143bb749572e080a4922a2c9d00000000000000000000000000000000000000000000000000000000000f69b51ca0239bfaea0e564e825136ef60e1964e44a2cf4b870a243acd447c9a56126b7ebea043e74bae18396f906741418febe29e7ac611bfcca80846b311b1f986cd6ff747"]}
```

**Comment by obscuren** [comment]
Thanks. The function that decodes hex does not explicitly check for `0x`, it assumes the input is hex without the prefix.

@bas-vk wasn't this addressed in the new RPC?

**Comment by bas-vk** [comment]
The new RPC layer uses the same hex decode function as the old RPC layer and thus has the same semantics. This probably has been since the inception of the RPC layer. I will create a PR that addresses this. It also falsely accepts input of uneven length.

**Comment by ethers** [comment]
Hi, when can this be fixed please?  It's causing the community cycles like:
https://github.com/ConsenSys/truffle/issues/179
https://github.com/ethcore/parity/issues/1317

**Comment by karalabe** [comment]
Fixed in yesterday's (1.5.4) release.

**Comment by fjl** [comment]
This is not fixed yet.
